src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -691,9 +691,17 @@ public override void VisitBinaryOperationSyntax(BinaryOperationSyntax syntax)
                     return operatorInfo.ReturnType;
                 }
 
+                string? additionalInfo = null;
+                if (operandType1.TypeKind is TypeKind.StringLiteral &&
+                    operandType2.TypeKind is TypeKind.StringLiteral &&
+                    syntax.Operator is BinaryOperator.Add)
+                {
+                    additionalInfo = "Use string interpolation instead.";
```

Comment:
Please could you add this as a public const to `DiagnosticBuilder.cs`? Trying to keep all the error message strings in once place for when we need to add localization.

Reply:
I think you need to include a space at the start of this string - otherwise it'll start immediately after the "."